src/pytorch_lightning/core/module.py

```diff
@@ -424,6 +424,7 @@ def log(
             )
 
         value = apply_to_collection(value, numbers.Number, self.__to_tensor)
+        value = apply_to_collection(value, torch.Tensor, torch.squeeze)
```

Comment:
I would suggest checking numel_1 before squeezing.

Also (here or in a follow-up), I would merge these 3 `apply_to_collection` calls

Reply:
done!